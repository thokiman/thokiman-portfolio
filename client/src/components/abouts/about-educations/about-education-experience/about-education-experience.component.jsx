import React from "react";
import EducationGrid from "../../about-commons/education-parts/education-grid/education-grid.component";
import {
  AboutEducationExperienceAttributes,
  AboutEducationExperienceCompany,
  AboutEducationExperienceContent,
  AboutEducationExperienceDescription,
  AboutEducationExperienceField,
  AboutEducationExperienceGrade,
  AboutEducationExperienceStartEndDate,
  AboutEducationExperienceTitle
} from "./about-education-experience.styles";


export const AboutEducationExperience = ({ lastExperience }) => {
  return (
    <AboutEducationExperienceContent>
      <AboutEducationExperienceTitle>Experience</AboutEducationExperienceTitle>
      {lastExperience.map(
        ({
          id,
          company,
          field,
          description,
          startDate,
          endDate,
          grade,
          imagePath,
        }) => {
          return (
            <div key={id}>
              <AboutEducationExperienceAttributes>
                <AboutEducationExperienceCompany>
                  {company}
                </AboutEducationExperienceCompany>
                <AboutEducationExperienceField>
                  {field}
                </AboutEducationExperienceField>
                <AboutEducationExperienceGrade>
                  {grade}
                </AboutEducationExperienceGrade>
                <AboutEducationExperienceStartEndDate>
                  {startDate} - {endDate}
                </AboutEducationExperienceStartEndDate>
                <AboutEducationExperienceDescription>
                  {description}
                </AboutEducationExperienceDescription>
              </AboutEducationExperienceAttributes>
              <EducationGrid imagePath={imagePath} />
            </div>
          );
        }
      )}
    </AboutEducationExperienceContent>
  );
};

export default AboutEducationExperience;
