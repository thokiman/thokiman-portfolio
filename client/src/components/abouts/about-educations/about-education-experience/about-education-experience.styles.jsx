import styled, { css } from "styled-components";

export const AboutEducationExperienceContent = styled.div`
  padding: 5px;
  display: flex;
  align-items: center;
  flex-direction: column;
  justify-content: center;
  width: 92%;
  margin: 0 25px;
  position: relative;
  top: 10px;
`;

export const AboutEducationExperienceTitle = styled.div`
  display: inline-block;
  width: 90%;
  font-size: 500%;
  text-align: left;
  line-height: 1.3em;
  letter-spacing: 0.04em;
  font-kerning: auto;
  padding: 0 2px;
  color: #e3e1e4;
  @media only screen and (max-width: 800px) {
    font-size: 17vw;
    width: 100%;
  }
`;

export const AboutEducationExperienceAttributes = styled.div`
  display: grid;
  grid-auto-columns: repeat(4, 1fr);
  grid-auto-rows: repeat(4, 1fr);
  gap: 0px 0px;
  grid-template-areas:
    "education-experience-company education-experience-company education-experience-field education-experience-field"
    "education-experience-company education-experience-company education-experience-description education-experience-description"
    "education-experience-start-end-date education-experience-start-end-date education-experience-description education-experience-description"
    "education-experience-grade education-experience-grade education-experience-description education-experience-description";
  width: 100%;
  height: 100%;
`;

const AboutEducationExperienceStyles = css`
  line-height: 1.3em;
  letter-spacing: 0.04em;
  font-kerning: auto;
  text-align: center;
  border-bottom: #74c6d4 solid 2px;
  color: #e3e1e4;
  display: flex;
  align-items: center;
  justify-content: center;
`;

export const AboutEducationExperienceCompany = styled.div`
  ${AboutEducationExperienceStyles}
  grid-area: education-experience-company;
  font-size: 250%;
  padding: 10px 0;
  @media only screen and (max-width: 800px) {
    padding: 7px;
    font-size: 8vw;
    text-align: left;
  }
`;

export const AboutEducationExperienceField = styled.div`
  ${AboutEducationExperienceStyles}
  grid-area: education-experience-field;
  font-size: 200%;
  padding: 0 10px;
  border-left: #74c6d4 solid 2px;
  @media only screen and (max-width: 800px) {
    padding: 7px;
    font-size: 7vw;
  }
`;

export const AboutEducationExperienceStartEndDate = styled.div`
  ${AboutEducationExperienceStyles}
  grid-area: education-experience-start-end-date;
  padding: 10px 0;
  font-size: 150%;
  @media only screen and (max-width: 800px) {
    padding: 4px;
    font-size: 5vw;
  }
`;

export const AboutEducationExperienceDescription = styled.div`
  ${AboutEducationExperienceStyles}
  grid-area: education-experience-description;
  padding: 10px;
  text-align: left;
  font-size: 120%;
  border-left: #74c6d4 solid 2px;
  @media only screen and (max-width: 800px) {
    padding: 7px;
    font-size: 4.5vw;
  }
`;

export const AboutEducationExperienceGrade = styled.div`
  ${AboutEducationExperienceStyles}
  grid-area: education-experience-grade;
  padding: 10px 0;
  font-size: 150%;
  @media only screen and (max-width: 800px) {
    padding: 4px;
    font-size: 5vw;
  }
`;

AboutEducationExperienceContent.displayName = "AboutEducationExperienceContent";
AboutEducationExperienceTitle.displayName = "AboutEducationExperienceTitle";
AboutEducationExperienceAttributes.displayName =
  "AboutEducationExperienceAttributes";
AboutEducationExperienceCompany.displayName = "AboutEducationExperienceCompany";
AboutEducationExperienceField.displayName = "AboutEducationExperienceField";
AboutEducationExperienceStartEndDate.displayName =
  "AboutEducationExperienceStartEndDate;";
AboutEducationExperienceDescription.displayName =
  "AboutEducationExperienceDescription";
AboutEducationExperienceGrade.displayName = "AboutEducationExperienceGrade";
