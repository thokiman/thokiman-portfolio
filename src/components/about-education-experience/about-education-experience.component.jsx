import React from "react";

import EducationGrid from "../education-grid/education-grid.component";
import "./about-education-experience.styles.scss";

const AboutEducationExperience = ({ lastExperience }) => {
  return (
    <div className="education-experience-content">
      <div className="education-experience-title">Experience</div>
      {lastExperience.map(
        ({
          id,
          company,
          field,
          description,
          startDate,
          endDate,
          grade,
          imagePath,
        }) => {
          return (
            <div key={id}>
              <div className="education-experience-attributes">
                <div className="education-experience-company">{company}</div>
                <div className="education-experience-field">{field}</div>
                <div className="education-experience-grade">{grade}</div>
                <div className="education-experience-start-end-date">
                  {startDate} - {endDate}
                </div>
                <div className="education-experience-description">
                  {description}
                </div>
              </div>
              <div className="education-experience-images">
                <EducationGrid imagePath={imagePath} />
              </div>
            </div>
          );
        }
      )}
    </div>
  );
};

export default AboutEducationExperience;
