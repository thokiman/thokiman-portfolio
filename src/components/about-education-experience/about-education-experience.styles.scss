.education-experience-content {

    padding: 5px;
    display: flex;
    align-items: center;
    flex-direction: column;
    justify-content: center;
    width: 92%;

    margin: 0 25px;
    position: relative;
    top: 10px;

    .education-experience-title {

        display: inline-block;
        width: 90%;
        font-size: 500%;
        text-align: left;
        line-height: 1.3em;
        letter-spacing: 0.04em;
        font-kerning: auto;
        padding: 0 2px;
        color: #e3e1e4
    }

    .education-experience-attributes {


        display: grid;
        grid-auto-columns: repeat(4, 1fr);
        grid-auto-rows: repeat(4, 1fr);
        gap: 0px 0px;
        grid-template-areas:
            "education-experience-company education-experience-company education-experience-field education-experience-field"
            "education-experience-company education-experience-company education-experience-description education-experience-description"
            "education-experience-start-end-date education-experience-start-end-date education-experience-description education-experience-description"
            "education-experience-grade education-experience-grade education-experience-description education-experience-description";
        width: 100%;
        height: 100%;


        .education-experience-company {
            grid-area: education-experience-company;
            padding: 10px 0;
            font-size: 250%;
            text-align: center;
            line-height: 1.3em;
            letter-spacing: 0.04em;
            font-kerning: auto;
            border-bottom: #74c6d4 solid 2px;
            color: #e3e1e4;
            display: flex;
            align-items: center;
            justify-content: center;

        }

        .education-experience-field {

            grid-area: education-experience-field;
            font-size: 200%;
            line-height: 1.3em;
            letter-spacing: 0.04em;
            font-kerning: auto;
            text-align: center;
            padding: 0 10px;
            border-left: #74c6d4 solid 2px;
            border-bottom: #74c6d4 solid 2px;
            color: #e3e1e4;
            display: flex;
            align-items: center;
            justify-content: center;


        }

        .education-experience-start-end-date {

            grid-area: education-experience-start-end-date;
            padding: 10px 0;
            font-size: 150%;
            text-align: center;
            line-height: 1.3em;
            letter-spacing: 0.04em;
            font-kerning: auto;
            border-bottom: #74c6d4 solid 2px;
            color: #e3e1e4;
            display: flex;
            align-items: center;
            justify-content: center;


        }

        .education-experience-description {

            grid-area: education-experience-description;
            padding: 10px 0;

            font-size: 120%;
            text-align: center;
            line-height: 1.3em;
            letter-spacing: 0.04em;
            font-kerning: auto;
            border-left: #74c6d4 solid 2px;
            border-bottom: #74c6d4 solid 2px;
            color: #e3e1e4;
            display: flex;
            align-items: center;
            justify-content: center;


        }

        .education-experience-grade {

            grid-area: education-experience-grade;
            padding: 10px 0;
            font-size: 150%;
            text-align: center;
            line-height: 1.3em;
            letter-spacing: 0.04em;
            font-kerning: auto;
            border-bottom: #74c6d4 solid 2px;
            color: #e3e1e4;
            display: flex;
            align-items: center;
            justify-content: center;




        }

    }

    .education-experience-images {}
}