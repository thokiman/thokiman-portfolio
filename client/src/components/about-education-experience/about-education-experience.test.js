import React from "react";
import { shallow } from "enzyme";

import { AboutEducationExperience } from "./about-education-experience.component";

describe("AboutEducationExperience component", () => {
  let wrapper;
  const mockArray = [];

  beforeEach(() => {
    const mockProps = {
      lastExperience: mockArray,
    };
    wrapper = shallow(<AboutEducationExperience {...mockProps} />);
  });

  it("should render AboutEducationExperience component", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
