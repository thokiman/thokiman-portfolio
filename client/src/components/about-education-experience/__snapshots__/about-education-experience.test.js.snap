// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutEducationExperience component should render AboutEducationExperience component 1`] = `ShallowWrapper {}`;
